src/Core/src/Platform/Android/MauiAppCompatActivity.cs

```diff
@@ -1,63 +1,149 @@
+		IWindow? _window;
+		Bundle? _savedInstanceState;
+
+		AndroidApplicationLifecycleState _currentState;
+		AndroidApplicationLifecycleState _previousState;
+
+		public MauiAppCompatActivity()
+		{
+			_previousState = AndroidApplicationLifecycleState.Uninitialized;
+			_currentState = AndroidApplicationLifecycleState.Uninitialized;
+		}
+
 		protected override void OnCreate(Bundle? savedInstanceState)
 		{
 			base.OnCreate(savedInstanceState);
 
-			if (App.Current as MauiApp == null)
-				throw new InvalidOperationException($"App is not {nameof(MauiApp)}");
+			_savedInstanceState = savedInstanceState;
 
-			var mauiApp = (MauiApp)App.Current;
+			if (MauiApplication.Current == null)
+				throw new InvalidOperationException($"App is not {nameof(Application)}");
 
-			if (mauiApp.Services == null)
-				throw new InvalidOperationException("App was not initialized");
+			_app = MauiApplication.Current;
 
-			var mauiContext = new MauiContext(mauiApp.Services, this);
+			if (_app == null || _app.Services == null)
+				throw new InvalidOperationException("App was not intialized");
+
+			_previousState = _currentState;
+			_currentState = AndroidApplicationLifecycleState.OnCreate;
+
+			var mauiContext = new MauiContext(_app.Services, this);
 			var state = new ActivationState(mauiContext, savedInstanceState);
-			var window = mauiApp.CreateWindow(state);
+			_window = _app.CreateWindow(state);
 
-			window.MauiContext = mauiContext;
+			_window.MauiContext = mauiContext;
 
-			//Hack for now we set this on the App Static but this should be on IFrameworkElement
-			App.Current.SetHandlerContext(window.MauiContext);
+			// Hack for now we set this on the App Static but this should be on IFrameworkElement
+			MauiApplication.Current?.SetHandlerContext(_window.MauiContext);
 
-			var content = (window.Page as IView) ?? 
-				window.Page.View;
+			var content = (_window.Content as IView) ??
+				_window.Content?.View;
 
 			CoordinatorLayout parent = new CoordinatorLayout(this);
 
-			SetContentView(parent, new ViewGroup.LayoutParams(CoordinatorLayout.LayoutParams.MatchParent, CoordinatorLayout.LayoutParams.MatchParent));
+			SetContentView(parent, new ViewGroup.LayoutParams(ViewGroup.LayoutParams.MatchParent, ViewGroup.LayoutParams.MatchParent));
+
+			parent.AddView(content?.ToNative(_window.MauiContext), new CoordinatorLayout.LayoutParams(ViewGroup.LayoutParams.MatchParent, ViewGroup.LayoutParams.MatchParent));
+		}
+				
+		protected override void OnStart()
+		{
+			base.OnStart();
+
+			_previousState = _currentState;
+			_currentState = AndroidApplicationLifecycleState.OnStart;
 
-			//AddToolbar(parent);
+			UpdateApplicationLifecycleState();
+		}
+
+		protected override void OnPause()
+		{
+			base.OnPause();
 
-			parent.AddView(content.ToNative(window.MauiContext), new CoordinatorLayout.LayoutParams(CoordinatorLayout.LayoutParams.MatchParent, CoordinatorLayout.LayoutParams.MatchParent));
+			_previousState = _currentState;
```

Comment:
Can we move all this to a ActivityLifeTimeService or a ActivityLifeTime manager ?